tests/test_cases/test_array_buses/test_array_buses.py

```diff
@@ -0,0 +1,68 @@
+import cocotb
+from cocotb.clock import Clock
+from cocotb.drivers import BusDriver
+from cocotb.monitors import BusMonitor
+from cocotb.triggers import RisingEdge
+
+
+class TestDriver(BusDriver):
+    _signals = ["data", "valid"]
+
+    def __init__(self, entity, name, clock, **kwargs):
+        BusDriver.__init__(self, entity, name, clock, **kwargs)
+        self.bus.valid <= 0
+
+    @cocotb.coroutine
+    def _driver_send(self, transaction, sync=True):
+        clkedge = RisingEdge(self.clock)
+        if sync:
+            yield clkedge
+
+        self.log.info("Sending {}".format(transaction))
+        self.bus.valid <= 1
+        self.bus.data <= transaction
+
+        yield clkedge
+        self.bus.valid <= 0
+
+
+class TestMonitor(BusMonitor):
+
+    _signals = ["data", "valid"]
+
+    def __init__(self, entity, name, clock, bank=0, **kwargs):
+        BusMonitor.__init__(self, entity, name, clock, **kwargs)
+        self.bank = bank
+        self.add_callback(self._print_res)
+
+    @cocotb.coroutine
+    def _monitor_recv(self):
+        clkedge = RisingEdge(self.clock)
+        while True:
+            yield clkedge
+            if self.bus.valid.value:
+                self._recv(int(self.bus.data))
+
+    def _print_res(self, transaction):
+        self.log.info("Received {} on bank {}".format(transaction, self.bank))
```

Comment:
The way this test is currently written means it can't actually fail. Can you append `transaction` to a list like `self.transactions`, and add an assert on the value of `self.transactions` at the end of `test_array_buses`?